packages/indexing/src/tables-manager.ts

```diff
@@ -238,44 +254,35 @@ export class PostgresTablesManager extends TablesManager {
   /**
    * Create mid tables and corresponding indexes
    */
-  override async initMidTables(modelsToIndex: Array<IndexModelArgs>) {
+  override async initMidTables(
+    modelsToIndex: Array<IndexModelArgs>,
+    createIndices: CreateIndicesRecord
+  ) {
     await Promise.all(
       modelsToIndex.map(async (modelIndexArgs) => {
-        await this.initMidTable(modelIndexArgs)
+        await this.initMidTable(modelIndexArgs, createIndices[modelIndexArgs.model.toString()])
       })
     )
   }
 
   /**
    * Create a single mid table for a given model
    */
-  async initMidTable(modelIndexArgs: IndexModelArgs) {
+  async initMidTable(modelIndexArgs: IndexModelArgs, createIndices: Array<FieldsIndex> = []) {
     const tableName = asTableName(modelIndexArgs.model)
     if (tableName.length > 63) {
       const errStr = `Cannot index model ${modelIndexArgs.model.toString()}.  Table name is too long: ${tableName}`
       this.logger.err(errStr)
       throw new Error(errStr)
     }
 
-    const relationColumns = relationsDefinitionsToColumnInfo(modelIndexArgs.relations)
     const exists = await this.dataSource.schema.hasTable(tableName)
     if (!exists) {
       this.logger.imp(`Creating ComposeDB Indexing table for model: ${tableName}`)
-      await createPostgresModelTable(this.dataSource, tableName, relationColumns)
-      if (modelIndexArgs.indices) {
-        await createPostgresIndices(this.dataSource, tableName, modelIndexArgs.indices)
-      }
-    } else if (relationColumns.length) {
-      // Make relations columns nullable
-      await this.dataSource.schema.alterTable(tableName, (table) => {
-        for (const column of relationColumns) {
-          if (column.type === ColumnType.STRING) {
-            const columnName = addColumnPrefix(column.name)
-            table.string(columnName, 1024).nullable().alter()
-          }
-        }
-      })
+      await createPostgresModelTable(this.dataSource, tableName)
     }
+
+    await createPostgresIndices(this.dataSource, tableName, createIndices)
```

Comment:
Given there's a promise involved here, it would be better to have this be only called if there's indices to create.